### PR TITLE
Feat/add video

### PR DESCRIPTION
**Pull request by James-cloudcannon**
# Context

Adds a generic responsive video component that can be used within other components. A user can choose between youtube or vimeo which is set within the `video_source` key. The user can then enter the link into the `video_url` key.

[Link to Notion ticket](https://www.notion.so/cloudcannon/Video-a65cab5c974e4efba24f83851108514a?pvs=4)

# Testing (tester)

The reviewer should check on this branch:

- changing the video source via the `video_source` key
-  adding a link to the `video_url` key.
- The component in Bookshop browser ( generic/video )
- [The site in CloudCannon (link here)](https://app.cloudcannon.com/42158/editor#sites/115950/dashboard/summary)

# Before merge (PR owner)

- Delete the test site in CloudCannon

### CODE REVIEW

**Review by aleksandrgordeev — APPROVED** [review]
Works well 👍. The only thing that is missing is the image parameter. I’m not sure if it’s a requirement or not.

Approved considering that either image will be added or ticket description is fixed to remove this parameter.

**Comment by James-cloudcannon** [comment]
> Works well 👍. The only thing that is missing is the image parameter. I’m not sure if it’s a requirement or not.
> 
> Approved considering that either image will be added or ticket description is fixed to remove this parameter.

Fallback image added. Test by adding an image to to image.src and removing the video_url

**Comment by aleksandrgordeev** [comment]
Thanks for the changes, @James-cloudcannon. Works well 👍

File: component-library/components/generic/video/video.bookshop.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  label: "Video"
+  description: "Simple video component"
+  icon: ondemand_videogit
+  tags: []
+
+# Defines the structure of this component, as well as the default values
+blueprint:
+  video_source:
+  video_url:
+  image:
```
Comment (jtrbuhovic) [inline component-library/components/generic/video/video.bookshop.yml]:
```suggestion
  fallback_image:
```

File: component-library/components/generic/video/video.bookshop.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    src:
```
Comment (jtrbuhovic) [inline component-library/components/generic/video/video.bookshop.yml]:
```suggestion
    image_source: 
```

File: component-library/components/generic/video/video.bookshop.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  icon: ondemand_videogit
+  tags: []
+
+# Defines the structure of this component, as well as the default values
+blueprint:
+  video_source:
+  video_url:
+  image:
+    src: 
+    alt:
```
Comment (jtrbuhovic) [inline component-library/components/generic/video/video.bookshop.yml]:
```suggestion
    alt_description: 
```

File: component-library/components/generic/video/video.bookshop.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  video_url:
+  image:
+    src: 
+    alt: 
+
+# Overrides any fields in the blueprint when viewing this component in the component browser
+preview:
+  video_source: "YouTube"
+  video_url: https://www.youtube.com/watch?v=LeqKIzeZel4
+  image:
```
Comment (jtrbuhovic) [inline component-library/components/generic/video/video.bookshop.yml]:
```suggestion
  fallback_image: 
```

File: component-library/components/generic/video/video.bookshop.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    src: 
+    alt: 
+
+# Overrides any fields in the blueprint when viewing this component in the component browser
+preview:
+  video_source: "YouTube"
+  video_url: https://www.youtube.com/watch?v=LeqKIzeZel4
+  image: 
+    src: https://placekitten.com/1000
```
Comment (jtrbuhovic) [inline component-library/components/generic/video/video.bookshop.yml]:
```suggestion
    image_source: https://placekitten.com/1000
```

File: component-library/components/generic/video/video.bookshop.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    alt: "A placeholder image"
```
Comment (jtrbuhovic) [inline component-library/components/generic/video/video.bookshop.yml]:
```suggestion
    alt_description: "A placeholder image"
```

File: component-library/components/generic/video/video.bookshop.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+_inputs:
+  video_source: 
+    type: select 
+    options:
+      values:
+        - "YouTube"
+        - "Vimeo"
+  video_url: 
+    type: url 
+  src:
```
Comment (jtrbuhovic) [inline component-library/components/generic/video/video.bookshop.yml]:
```suggestion
  image_source: 
```

File: component-library/components/generic/video/video.eleventy.liquid [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    <div class="{{ c }}__video-aspect-wrapper">
+      <div class="{{ c }}__video">
+        <iframe class="{{ c }}__youtube" src="{{ url }}" width="100%" height="100%" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
+      </div>
+    </div>
+  {% endif %}
+{% else %}
+  <div class="{{ c }}__video-aspect-wrapper">
+    <div class="{{ c }}__video">
+      <img class="{{ c }}__default-image" src="{{ image.src }}" alt="{{ image.alt }}">
```
Comment (jtrbuhovic) [inline component-library/components/generic/video/video.eleventy.liquid]:
```suggestion
      <img class="{{ c }}__default-image" src="{{ fallback_image.image_source }}" alt="{{ fallback_image.alt_description }}">
```

**Review by jtrbuhovic — CHANGES_REQUESTED** [review]
Looks great! I made some suggestions to the fallback image input. "src", "alt" etc. probably don't mean much to an editor so I think it's best to be verbose when naming inputs/make them self documenting
Screenshot 2023-09-29 at 4 32 02 PM

**Comment by AlyshaNolan** [comment]
I question the need for this to have a fallback image? The user should never use this component and not provide a video (they should then use the image component). Alternatively, I think that the fallback image should have more meaning and be used while the video is loading or if the link is invalid and fails to load (if that check is possible?)

**Comment by jtrbuhovic** [comment]
@AlyshaNolan, @James-cloudcannon I've moved the image so it works as a thumbnail/loads before the video

**Review by AlyshaNolan — APPROVED** [review]

**Pull request merged**